filebeat/input/udp/input.go

```diff
@@ -18,63 +18,90 @@
 package udp
 
 import (
-	"fmt"
-	"sync"
+	"net"
 	"time"
 
-	"github.com/mitchellh/hashstructure"
+	"github.com/dustin/go-humanize"
 	"github.com/rcrowley/go-metrics"
 
-	"github.com/elastic/beats/v7/filebeat/channel"
-	"github.com/elastic/beats/v7/filebeat/harvester"
-	"github.com/elastic/beats/v7/filebeat/input"
+	input "github.com/elastic/beats/v7/filebeat/input/v2"
+	stateless "github.com/elastic/beats/v7/filebeat/input/v2/input-stateless"
 	"github.com/elastic/beats/v7/filebeat/inputsource"
 	"github.com/elastic/beats/v7/filebeat/inputsource/udp"
 	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/beats/v7/libbeat/feature"
 	"github.com/elastic/beats/v7/libbeat/monitoring/inputmon"
 	conf "github.com/elastic/elastic-agent-libs/config"
-	"github.com/elastic/elastic-agent-libs/logp"
 	"github.com/elastic/elastic-agent-libs/mapstr"
 	"github.com/elastic/elastic-agent-libs/monitoring"
 	"github.com/elastic/elastic-agent-libs/monitoring/adapter"
+	"github.com/elastic/go-concert/ctxtool"
 )
 
-func init() {
-	err := input.Register("udp", NewInput)
-	if err != nil {
-		panic(err)
+func Plugin() input.Plugin {
+	return input.Plugin{
+		Name:       "udp",
+		Stability:  feature.Stable,
+		Deprecated: false,
+		Info:       "udp packet server",
+		Manager:    stateless.NewInputManager(configure),
 	}
 }
 
-// Input defines a udp input to receive event on a specific host:port.
-type Input struct {
-	sync.Mutex
-	udp     *udp.Server
-	started bool
-	outlet  channel.Outleter
+func configure(cfg *conf.C) (stateless.Input, error) {
+	config := defaultConfig()
+	if err := cfg.Unpack(&config); err != nil {
+		return nil, err
+	}
 
-	metrics *inputMetrics
+	return newServer(config)
 }
 
-// NewInput creates a new udp input
-func NewInput(cfg *conf.C, outlet channel.Connector, context input.Context) (input.Input, error) {
-	id, err := configID(cfg)
-	if err != nil {
-		return nil, err
+func defaultConfig() config {
+	return config{
+		Config: udp.Config{
+			MaxMessageSize: 10 * humanize.KiByte,
+			// TODO: What should be default port?
```

Comment:
Let's remove the TODOs. I think these have been "decided" given how long they have been the defaults.